plugins/base/src/main/kotlin/renderers/gfm/GfmRenderer.kt

```diff
@@ -0,0 +1,185 @@
+
+    private fun StringBuilder.buildListItem(items: List<ContentNode>, pageContext: ContentPage, bullet: String = "*") {
+        items.forEach {
+            if(it is ContentList) {
+                val builder = StringBuilder()
+                builder.append(indent)
+                builder.buildListLevel(it, pageContext)
+                append(builder.toString().replace(Regex("  \n(?!$)"), "  \n$indent"))
+            } else {
+                append("$bullet ")
+                it.build(this, pageContext)
+                buildNewLine()
+            }
+        }
+    }
+
+    private fun StringBuilder.buildListLevel(node: ContentList, pageContext: ContentPage) {
+        if(node.ordered) {
+            buildListItem(node.children, pageContext, "${node.start}.")
+        } else {
+            buildListItem(node.children, pageContext, "*")
+        }
+    }
+
+    override fun StringBuilder.buildNewLine() {
+        this.append("  \n")
+    }
+
+    fun StringBuilder.buildParagraph() {
+        this.append("\n\n")
+    }
+
+    override fun StringBuilder.buildResource(node: ContentEmbeddedResource, pageContext: ContentPage) {
+        this.append("Resource")
+    }
+
+    override fun StringBuilder.buildTable(node: ContentTable, pageContext: ContentPage) {
+
+        val size = node.children.firstOrNull()?.children?.size ?: 0
+
+        buildParagraph()
+
+        if(node.header.size > 0) {
+            node.header.forEach {
+                it.children.forEach {
+                    append("| ")
+                    it.build(this, pageContext)
+                }
+                append("|\n")
+            }
+        } else {
+            append("| ".repeat(size))
+            if(size > 0) append("|\n")
+        }
+
+        append("|---".repeat(size))
+        if(size > 0) append("|\n")
+
+
+        node.children.forEach {
+            it.children.forEach {
+                append("| ")
+                it.build(this,  pageContext)
+            }
+            append("|\n")
+        }
+
+        buildParagraph()
+    }
+
+    private fun String.escapeIllegalCharacters() = this
+        .replace("<", "\\<")
+        .replace(">", "\\>")
+
+    override fun StringBuilder.buildText(textNode: ContentText) {
+        val decorators = decorators(textNode.style)
+        this.append(decorators)
+        this.append(textNode.text.escapeIllegalCharacters())
+        this.append(decorators.reversed())
+    }
+
+    override fun StringBuilder.buildNavigation(page: PageNode) {
+        locationProvider.ancestors(page).asReversed().forEach { node ->
+            append("/")
+            if (node.isNavigable) buildLink(node, page)
+            else append(node.name)
+        }
+        buildParagraph()
+    }
+
+    override fun buildPage(page: ContentPage, content: (StringBuilder, ContentPage) -> Unit): String {
+        val builder = StringBuilder()
+        content(builder, page)
+        return builder.toString()
+    }
```

Comment:
```suggestion
    override fun buildPage(page: ContentPage, content: (StringBuilder, ContentPage) -> Unit) =
        content(StringBuilder(), page).toString()
```

Reply:
It's not a content() we won't call toString() but StringBuilder object.

Reply:
In my opinion there is barely anything we can do to zip it. Either way we need to keep reference to `StringBuilder` to have possibility to call later `toString()` on it.

Reply:
I just come up with 
```    
override fun buildPage(page: ContentPage, content: (StringBuilder, ContentPage) -> Unit): String =
        StringBuilder().let {
            content(it, page)
            it
        }.toString()
```
Both version suits me so decide which one you prefer

Reply:
I just come up with 
```    
override fun buildPage(page: ContentPage, content: (StringBuilder, ContentPage) -> Unit): String =
        StringBuilder().apply {
            content(this, page)
        }.toString()
```
Of course we can use also instead
Both version suits me so decide which one you prefer